s3.tf

```diff
@@ -1,7 +1,10 @@
+resource "random_uuid" "cloudtrail_bucket_name" {
+}
+
 # temporarily ignoring because tfsec does not yet support AWS v4 configuration
 # tfsec:ignore:aws-s3-enable-bucket-encryption tfsec:ignore:aws-s3-encryption-customer-key tfsec:ignore:aws-s3-enable-bucket-logging tfsec:ignore:aws-s3-enable-versioning
 resource "aws_s3_bucket" "cloudtrail_bucket" {
-  bucket_prefix = var.prefix
+  bucket = "${var.prefix}-${random_uuid.cloudtrail_bucket_name.result}"
```

Comment:
I'd rather we stick with the bucket prefix and just `*` the condition on the key. If you see a bucket name suffixed with random text, you'd assume it was generated via a bucket_prefix, so to do it in such a custom way could lead to confusion.

And when we use a * in the condition in the KMS key, we can just comment why it's not specific. Due to the ordering of resources being created.

Reply:
bucket `prefix` is not even a thing in AWS; it's a tf construct. if `bucket` is not passed in it'll concatenate the prefix with a random number(probably timestamp) since bucket name needs to be globally unique. also i think i'd rather be stricter with the policy condition.